internal/provider/integration_gitlab_resource.go

```diff
@@ -0,0 +1,302 @@
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^https?:\/\/[a-zA-Z0-9\-._~:\/?#[\]@!$&'()*+,;=%]+$`),
+						"must be a valid URL",
+					),
+				},
+			},
+			"discovery": schema.SingleNestedAttribute{
+				Optional: true,
+				Attributes: map[string]schema.Attribute{
+					"groups": schema.BoolAttribute{
+						MarkdownDescription: "Enable discovery of GitLab groups.",
+						Optional:            true,
+					},
+					"projects": schema.BoolAttribute{
+						MarkdownDescription: "Enable discovery of GitLab projects.",
+						Optional:            true,
+					},
+					"terraform": schema.BoolAttribute{
+						MarkdownDescription: "Enable discovery of Terraform configurations.",
+						Optional:            true,
+					},
+					"k8s_manifests": schema.BoolAttribute{
+						MarkdownDescription: "Enable discovery of Kubernetes manifests.",
+						Optional:            true,
+					},
+				},
+			},
+			"credentials": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"token": schema.StringAttribute{
+						MarkdownDescription: "Token for GitLab integration.",
+						Required:            true,
+						Sensitive:           true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *integrationGitlabResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*ExtendedGqlClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *integrationGitlabResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data integrationGitlabResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Compute and validate the space
+	space, err := r.client.ComputeSpace(data.SpaceID)
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid Configuration", err.Error())
+		return
+	}
+	ctx = tflog.SetField(ctx, "space_mrn", space.MRN())
+
+	gitlabType := mondoov1.GitlabIntegrationType("NONE")
+	if data.Group.ValueString() != "" {
+		gitlabType = mondoov1.GitlabIntegrationType("GROUP")
+	}
+	// Create API call logic
+	tflog.Debug(ctx, "Creating integration")
+	integration, err := r.client.CreateIntegration(ctx,
+		space.MRN(),
+		data.Name.ValueString(),
+		mondoov1.ClientIntegrationTypeGitLab,
+		mondoov1.ClientIntegrationConfigurationInput{
+			GitLabConfigurationOptions: &mondoov1.GitlabConfigurationOptionsInput{
```

Comment:
nit; we normally have a function attached to the model, in this case `integrationGitlabResourceModel.GetConfigurationOptions()`, and there is where we load the config

Reply:
I moved config like you suggested